shared/src/bundles/users/types/user-sign-up-response-dto.type.ts

```diff
@@ -1,6 +1,7 @@
 type UserSignUpResponseDto = {
     id: number;
     email: string;
+    token?: string;
```

Comment:
Why token has optional type like string | undefined ?

Reply:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/26dfa73b-0bae-4906-a6fe-4afef9450087)

Reply:
I got it. Leave only token in UserSignUpResponseDto, for userService.create change return type to UserEntity and return UserEntity not an object

Reply:
> Why token has optional type like string | undefined ?

I created a new Dto for userService.create():
```
type UserCreationResponseDto = {
    id: number;
    email: string;
};

export { type UserCreationResponseDto };
```

Then extended that for UserResponseDto:
```
import { type UserCreationResponseDto } from "./user-creation-response-dto.type.js";

type UserSignUpResponseDto = UserCreationResponseDto & { token: string };

export { type UserSignUpResponseDto };
```

Reply:
> I got it. Leave only token in UserSignUpResponseDto, for userService.create change return type to UserEntity and return UserEntity not an object

I read this one late. Making changes now.

Reply:
> I got it. Leave only token in UserSignUpResponseDto, for userService.create change return type to UserEntity and return UserEntity not an object

These changes have been made.